tilelang/utils/pass_diff.py

```
@@ -0,0 +1,1060 @@
+    import time
+
+    total_add = sum(s["insertions"] for s in steps)
+    total_del = sum(s["deletions"] for s in steps)
+    changed_passes = sum(1 for s in steps if s["changed"])
+
+    html_steps = []
+    for idx, step in enumerate(steps):
+        changed = step["changed"]
+        badge_class = "changed" if changed else "unchanged"
+        badge_text = f"+{step['insertions']}/-{step['deletions']}" if changed else "unchanged"
+        stats_text = f"+{step['insertions']} −{step['deletions']}" if changed else ""
```

Ruff (0.15.15)

[warning] String contains ambiguous `−` (MINUS SIGN). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)